### DISCUSSION TITLE
typesetdir

### DISCUSSION THREAD

**Discussion by hansonchar** [latex3/l3build#390] [Q&A] [answered]
Reading l3build.pdf:
> `l3build doc`: Compiles documentation files in the typesetdir directory. In the absence of one or more
file names, all documentation is typeset;

In my `build.lua`, the compilation works if I have:
```lua
typesetfiles  = {"*.tex"}
typesetexe = "pdftex"
```
but if I moved the `*.tex` files to a `doc` subdirectory, and modified `build.lua` to become:
```lua
typesetdir = "doc"
-- typesetfiles  = {"*.tex"}
typesetexe = "pdftex"
```
`l3build doc` seems to delete all the files under "doc" without compiling any of them.

What am I missing?  (I am using l3build Release 2024-10-16.)

**Comment by josephwright** [comment, marked as answer]
You need something like `docfiledir = "./doc"`, i.e. a path

**Comment by hansonchar** [comment]
Experiments show that this works:
```
docfiledir = "doc"
typesetfiles  = {"*.tex"}
```
Great!

However,  specifying `docfiledir` *alone* doesn't work, and specifying `typesetdir` (regardless of whether `typesetfiles` is specified or not) would cause the files under that directory to be deleted (without compilation).  This suggests the support of `typesetdir` is broken, or at least inconsistent with `l3build.pdf`.  Please let me know if I am mistaken.

Should I create an issue on this?

**Comment by hansonchar** [comment]
Oh, I think I see what's going on now.  The "typesetdir" is the working directory l3build use during the documentation compilation, nothing to do with the location of the documentation input/source files!

The l3build.pdf documentation is technically correct, but a bit misleading to newbies.  Let me create an issue, and try to submit a pull request to improve the doc.
